webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoCompleteReducer.js

```diff
@@ -0,0 +1,41 @@
+import Immutable from 'seamless-immutable';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETION_REQUEST,
+  AUTO_COMPLETION_SUCCESS,
+  AUTO_COMPLETION_FAILURE,
+} from './AutoCompleteConstants';
+
+const initialState = Immutable({
+  options: [],
+  status: 'PENDING',
+  error: null,
+  searchQuery: '',
+});
+
+export default (state = initialState, action) => {
+  const { payload, type } = action;
+  switch (type) {
+    case AUTO_COMPLETION_REQUEST:
+      return state
+        .set('error', null)
+        .set('status', STATUS.PENDING);
+
+    case AUTO_COMPLETION_SUCCESS:
+      return state
+        .set('options', Object.keys(payload)
+          .filter(k => !['query', 'url'].includes(k))
+          .map(k => payload[k]))
```

Comment:
IMHO it would much better to use selectors with memorization hete (reselect?), this probably will invoked many times.

Reply:
I've added `reselect` in [this PR](https://github.com/theforeman/foreman/pull/5778)